zerver/views/upload.py

```diff
@@ -140,6 +140,28 @@ def serve_local_file_unauthed(request: HttpRequest, token: str, filename: str) -
     return serve_local(request, path_id, url_only=False)
 
 
+def serve_local_avatar_unauthed(request: HttpRequest, path: str) -> HttpResponseBase:
+    if settings.LOCAL_UPLOADS_DIR is None:
+        # We do not expect to see accesses to this path when serving
+        # S3-based avatars; we expect clients to hit S3 directly.  We
+        # could potentially 302 to the appropriate S3 URL.
+        return HttpResponseNotFound("<p>File not found</p>")
+
+    assert settings.LOCAL_AVATARS_DIR is not None
+    if settings.DEVELOPMENT:
```

Comment:
Probably worth a comment noting here that in production, we use X-Accel-Redirect to have `nginx` serve the file, but the development environment doesn't run `nginx`, so we need to serve it directly.